core/src/dbs/group.rs

```diff
@@ -0,0 +1,386 @@
+			// Loop over each group clause
+			for group in groups.iter() {
+				// Get the value at the path
+				let val = obj.pick(group);
+				// Set the value at the path
+				arr.push(val);
+			}
+			// Add to grouped collection
+			match self.grp.get_mut(&arr) {
+				Some(agr) => Self::pushes(ctx, opt, txn, agr, &self.idioms, obj).await?,
+				None => {
+					let mut agr: Vec<Aggregator> =
+						self.base.iter().map(|a| a.new_instance()).collect();
+					Self::pushes(ctx, opt, txn, &mut agr, &self.idioms, obj).await?;
+					self.grp.insert(arr, agr);
+				}
+			}
+		}
+		Ok(())
+	}
+
+	async fn pushes(
+		ctx: &Context<'_>,
+		opt: &Options,
+		txn: &Transaction,
+		agrs: &mut [Aggregator],
+		idioms: &[Idiom],
+		obj: Value,
+	) -> Result<(), Error> {
+		for (agr, idiom) in agrs.iter_mut().zip(idioms) {
+			let val = obj.get(ctx, opt, txn, None, idiom).await?;
+			agr.push(val)?;
+		}
+		Ok(())
+	}
+
+	pub(super) fn len(&self) -> usize {
+		self.grp.len()
+	}
+
+	pub(super) async fn output(
+		&mut self,
+		ctx: &Context<'_>,
+		opt: &Options,
+		txn: &Transaction,
+		stm: &Statement<'_>,
+	) -> Result<StoreCollector, Error> {
+		let mut results = StoreCollector::default();
+		if let Some(fields) = stm.expr() {
+			let grp = mem::take(&mut self.grp);
+			// Loop over each grouped collection
+			for (_, mut aggregator) in grp {
+				// Create a new value
+				let mut obj = Value::base();
+				// Loop over each group clause
+				for field in fields.other() {
+					// Process the field
+					if let Field::Single {
+						expr,
+						alias,
+					} = field
+					{
+						let idiom = alias
+							.as_ref()
+							.map(Cow::Borrowed)
+							.unwrap_or_else(|| Cow::Owned(expr.to_idiom()));
+						if let Some(idioms_pos) =
+							self.idioms.iter().position(|i| i.eq(idiom.as_ref()))
+						{
+							let agr = &mut aggregator[idioms_pos];
+							match expr {
+								Value::Function(f) if f.is_aggregate() => {
+									let a = f.get_optimised_aggregate();
+									let x = if matches!(a, OptimisedAggregate::None) {
+										// The aggregation is not optimised, let's compute it with the values
+										let vals = agr.take();
+										let x = vals
+											.all()
+											.get(ctx, opt, txn, None, idiom.as_ref())
+											.await?;
+										f.aggregate(x).compute(ctx, opt, txn, None).await?
+									} else {
+										// The aggregation is optimised, just get the value
+										agr.compute(a)?
+									};
+									obj.set(ctx, opt, txn, idiom.as_ref(), x).await?;
+								}
+								_ => {
+									let vals = agr.take();
+									let x = vals.first();
+									// TODO Check why this seems to not be required anymore
+									// let x = if let Some(alias) = alias {
+									// 	let cur = (&x).into();
+									// 	alias.compute(ctx, opt, txn, Some(&cur)).await?
+									// } else {
+									// 	let cur = (&x).into();
+									// 	expr.compute(ctx, opt, txn, Some(&cur)).await?
+									// };
```

Comment:
@tobiemh It seems this part does not play any role. Am I missing something?

Reply:
These are the operations it was doing. Which is now properly handled by the new design.

```js
Vals: [
	{
		count: 1,
		country: 'AUD',
		year: 2021
	}
]
x: {
	count: 1,
	country: 'AUD',
	year: 2021
}
alias-x: 2021
Vals: [
	{
		count: 1,
		country: 'AUD',
		year: 2021
	}
]
x: {
	count: 1,
	country: 'AUD',
	year: 2021
}
expr-x: 'AUD'
```